src/main/java/hudson/plugins/promoted_builds/Promotion.java

```diff
@@ -316,10 +320,13 @@ protected void post2(BuildListener listener) throws Exception {
                 }
 
                 // tickle PromotionTriggers
-                for (AbstractProject<?,?> p : Jenkins.getInstance().getAllItems(AbstractProject.class)) {
-                    PromotionTrigger pt = p.getTrigger(PromotionTrigger.class);
-                    if (pt!=null)
-                        pt.consider(Promotion.this);
+                Jenkins jenkins = Jenkins.getInstance();
+                if(jenkins != null){
```

Comment:
This update is reasonable, but IMO it would be better to polish the code in a separate pull request. Does it really happens in JobDSL?